src/OpenSpaceToolkit/Physics/Environment/Utilitity/Eclipse.cpp

```diff
@@ -17,12 +22,211 @@ namespace environment
 namespace utilities
 {
 
-Array<Interval> eclipseIntervalsAtPosition(
-    const Interval& anAnalysisInterval, const Position& aPosition, const Environment& anEnvironment
+using ostk::core::type::Shared;
+using ostk::core::type::Size;
+
+// EclipsePhase implementation
+EclipsePhase::EclipsePhase(const Region& aRegion, const Interval& anInterval, const bool& anIsComplete)
+    : region(aRegion),
+      interval(anInterval),
+      is_complete(anIsComplete)
+{
+}
+
+String EclipsePhase::getRegionString() const
+{
+    switch (region)
+    {
+        case Region::UMBRA:
+            return "Umbra";
+        case Region::PENUMBRA:
+            return "Penumbra";
+        default:
+            throw ostk::core::error::runtime::Undefined("Eclipse phase region");
+    }
+}
+
+// Eclipse implementation
+Eclipse::Eclipse(
+    const Celestial& anOccultedCelestialObject,
+    const Celestial& anOccultingCelestialObject,
+    const Array<EclipsePhase>& somePhases
 )
+    : occultedCelestialObject_(anOccultedCelestialObject),
+      occultingCelestialObject_(anOccultingCelestialObject),
+      phases_(somePhases)
+{
+    if (phases_.isEmpty())
+    {
+        throw ostk::core::error::runtime::Wrong("Eclipse has no phases.");
```

Comment:
```suggestion
        throw ostk::core::error::Runtime("Eclipse has no phases.");
```
If we use Wrong, it will actually say `Eclipse has no Phases. is Wrong.`